datafusion/functions/src/datetime/to_local_time.rs

```diff
@@ -0,0 +1,601 @@
+        );
+
+        // TODO chunchun: invalid timestamp: us, ms, s
+    }
+
+    #[test]
+    fn test_to_local_time_scalar() {
+        let timezone = Some("Europe/Brussels".into());
+        let timestamps_with_timezone = vec![
+            (
+                ScalarValue::TimestampNanosecond(
+                    Some(1_123_123_000_000_000_000),
+                    timezone.clone(),
+                ),
+                ScalarValue::TimestampNanosecond(Some(1_123_130_200_000_000_000), None),
+            ),
+            (
+                ScalarValue::TimestampMicrosecond(
+                    Some(1_123_123_000_000_000),
+                    timezone.clone(),
+                ),
+                ScalarValue::TimestampMicrosecond(Some(1_123_130_200_000_000), None),
+            ),
+            (
+                ScalarValue::TimestampMillisecond(
+                    Some(1_123_123_000_000),
+                    timezone.clone(),
+                ),
+                ScalarValue::TimestampMillisecond(Some(1_123_130_200_000), None),
+            ),
+            (
+                ScalarValue::TimestampSecond(Some(1_123_123_000), timezone),
+                ScalarValue::TimestampSecond(Some(1_123_130_200), None),
+            ),
+        ];
+
+        for (input, expected) in timestamps_with_timezone {
+            test_to_local_time_helpfer(input, expected);
+        }
+    }
+
+    #[test]
+    fn test_timezone_with_daylight_savings() {
+        let timezone_str = "America/New_York";
+
+        let test_cases = vec![
+            // DST time
+            "2019-03-31T04:00:00",
+            "2020-03-31T13:40:00",
+            // End of DST
+            "2019-11-04T04:06:40",
+            "2020-11-04T14:06:40",
+        ];
+
+        for timestamp_str in test_cases {
+            // timestamp in local timezone
+            let tz: arrow::array::timezone::Tz =
+                timezone_str.parse().expect("Invalid timezone");
+            let timestamp = timestamp_str
+                .parse::<NaiveDateTime>()
+                .unwrap()
+                .and_local_timezone(tz) // this is in a local timezone
+                .unwrap()
+                .timestamp_nanos_opt()
+                .unwrap();
+
+            // timestamp with no timezone
+            let expected_timestamp = timestamp_str
+                .parse::<NaiveDateTime>()
+                .unwrap()
+                .and_utc() // this is in UTC
+                .timestamp_nanos_opt()
+                .unwrap();
+
+            let input = ScalarValue::TimestampNanosecond(
+                Some(timestamp),
+                Some(timezone_str.into()),
+            );
+            let expected =
+                ScalarValue::TimestampNanosecond(Some(expected_timestamp), None);
+            test_to_local_time_helpfer(input, expected)
+        }
+    }
+
+    fn test_to_local_time_helpfer(input: ScalarValue, expected: ScalarValue) {
```

Comment:
typo?

```suggestion
    fn test_to_local_time_helper(input: ScalarValue, expected: ScalarValue) {
```

Reply:
😅 fixed in https://github.com/apache/datafusion/pull/11347/commits/f8ef51b37aab9de38957a139c306783ef0500099